test/Storage.js

```diff
@@ -156,5 +156,127 @@ describe('Storage', () => {
           }
       });
     });
+  });
+
+  describe('listDataByDevice', () => {
+    let storage, query;
+    const deviceId = '0643ca1462f94b79';
+
+    describe('when deviceId is invalid', () => {
+      beforeEach(() => {
+        storage = new Storage(settings);
+      });
+
+      it('should throw an exception', async () => {
+        expect(storage.listDataByDevice('4797@$fa'))
+          .to.eventually.be.rejected;
+      });
+    });
+
+    describe('with empty query', () => {
+      beforeEach(() => {
+        storage = new Storage(settings);
+        nock(url)
+          .get(`/data/${deviceId}`)
+          .reply(200, dataList);
+      });
+
+      it('should returns an array', async () => {
+        const response = await storage.listDataByDevice(deviceId);
+        expect(response).to.be.an('array');
+      });
+
+      it('should returns an array of data objects', async () => {
+        const response = await storage.listDataByDevice(deviceId);
+        expect(response).to.deep.equal(dataList);
+      });
+    });
+
+    describe('with date interval query', () => {
```

Comment:
See discussion in previous patches about these tests.

Reply:
Waiting for response.

Reply:
Done.